### DISCUSSION TITLE
x-bind:value does not set value attribute for empty strings

### DISCUSSION THREAD

**Discussion by bmihelac** [alpinejs/alpine#2679] [7. Bugs (archived)] [answered]
Hello all, 

when `x-bind:value` is used with empty string, `value` attribute is not set in `option` element.

For example, in the following code, first option would not have value attribute at all. I have checked documentation but did not found anything that could explain this behavior, so I wonder if this is a bug. It looks like this is because empty string is falsy.

```html
<select x-model="sortBy">
    <template x-for="el in [['','Sort by'],['desc','Desc']]">
      <option :value="el[0]" x-text="el[1]"></option>
    </template>
</select>
```

I have created minimal use case:
https://codesandbox.io/s/alpinejs-x-value-r0w9sq?file=/index.html

Attributes other than `value` are added to an element even if they are empty string.

Workaround for the case above is to wrap string in String class.

**Comment by AndiLeni** [comment]
I stumbled across this same issue and can confirm it.

**Comment by ekwoka** [comment]
Hmm, yeah, that is incorrect.

since no value means the text content is used as the value, while an empty value is an empty value...

**Reply by hirasso** [reply]
Any news for this issue? Or an official workaround?

**Reply by ekwoka** [reply]
Oh, looking at the codesandbox, I see that that isn't the same as the native behavior where an empty value is an empty value attribute, so yeah, this would definitely be a bug. It should have the same end behavior as `value` as a bare attribute.

**Reply by hirasso** [reply, marked as answer]
Recently a PR was merged with a fix: https://github.com/alpinejs/alpine/pull/4831
